src/intelligence_layer/core/model.py

```diff
@@ -157,6 +157,22 @@ def __init__(
         )
         self._explain = _Explain(self._client, name)
 
+    @property
+    @lru_cache(maxsize=1)
```

Comment:
I recently looked into how this lru cache behaves. Apparently, this cache is on a class-level, meaning if you use two different models, the cache never hits if both models ask this context size in an A -> B -> A fashion. Is this what you intended?

Reply:
Well, as long as the same arguments are passed in, you will get the same output.
As self is different for different class instances, I believe that this will work on an instance level, right?

Reply:
```
from functools import lru_cache


class TestClass():
    def __init__(self, foo: str) -> None:
        self._foo = foo

    @lru_cache
    def foo(self) -> str:
        return self._foo
    

tc1 = TestClass("hello")
tc2 = TestClass("world")

print(tc1.foo())
print(tc2.foo())
```